### ISSUE TITLE
GlyphFactory.cpp:20:22: error: array subscript has type 'char' [-Werror=char-subscripts]

### ISSUE THREAD

**Issue by franz-ms-muc** [BartVandewoestyne/Design-Patterns-GoF#2]
GlyphFactory.cpp:20:22: error: array subscript has type 'char' \[-Werror=char-subscripts\]

char is signed on some machines. 

a Subscript of an Array is defined unsigned in C (0....n)

**Comment by franz-ms-muc** [comment]
[https://github.com/Meisterschulen-am-Ostbahnhof-Munchen/Design-Patterns-GoF-on-ESP32/blob/feature/1/Structural_Patterns/Flyweight/components/Flyweight_component/GlyphFactory.cpp](https://github.com/Meisterschulen-am-Ostbahnhof-Munchen/Design-Patterns-GoF-on-ESP32/blob/feature/1/Structural_Patterns/Flyweight/components/Flyweight_component/GlyphFactory.cpp)

here i will correct it so it runs in ESP32. please Double check it before take over to your Repo, because i do not check if it is running on PC !

**Comment by franz-ms-muc** [comment]
same Story for the delete\[\]. there is no new\[\]. so delete\[\] is wrong. 

https://stackoverflow.com/questions/2425728/delete-vs-delete-operators-in-c

**Comment by BartVandewoestyne** [comment]
> same Story for the delete[]. there is no new[]. so delete[] is wrong. 
> 
> https://stackoverflow.com/questions/2425728/delete-vs-delete-operators-in-c

Thanks. I fixed the `GlyphFactory` destructor in commit [d2641d8](https://github.com/BartVandewoestyne/Design-Patterns-GoF/commit/d2641d8099b5d1c05b70e68e6566d3ca45a4bd2d).

**Comment by BartVandewoestyne** [comment]
> GlyphFactory.cpp:20:22: error: array subscript has type 'char' [-Werror=char-subscripts]
> 
> char is signed on some machines. 
> 
> a Subscript of an Array is defined unsigned in C (0....n)

I agree that there is an issue with the code and that this warning is justifiable.  However, this is how the code in the book is.
I am thinking of keeping the book's code in the `master` branch and then creating an extra branch for all the bugfixes to the book's code.  Doing a diff between the `master` branch and that branch would then show where the book's code had some issues...
